lib/virt_autotest/utils.pm

```diff
@@ -137,31 +138,34 @@ sub print_cmd_output_to_file {
 }
 
 sub download_script_and_execute {
-    my (%args) = @_;
+    my ($script_name, %args) = @_;
     $args{output_file} //= "$args{script_name}.log";
     $args{machine} //= 'localhost';
 
-    download_script(script_name => $args{script_name}, script_url => $args{script_url}, machine => $args{machine});
-    my $cmd = "~/$args{script_name}";
+    download_script($script_name, script_url => $args{script_url}, machine => $args{machine});
+    my $cmd = "~/$script_name";
     $cmd = "ssh root\@$args{machine} " . "\"$cmd\"" if ($args{machine} ne 'localhost');
     script_run("$cmd >> $args{output_file} 2>&1");
 }
 
 sub download_script {
-    my (%args) = @_;
-    $args{script_name} //= '';
-    $args{script_url} //= data_url("virt_autotest/$args{script_name}");
-    $args{machine} //= 'localhost';
-
-    my $cmd = "curl -s -o ~/$args{script_name} $args{script_url}";
-    $cmd = "ssh root\@$args{machine} " . "\"$cmd\"" if ($args{machine} ne 'localhost');
-    my $ret = script_run($cmd);
-    unless ($ret == 0) {
-        record_info('Softfail', "Failed to download $args{script_name} from $args{script_url}!", result => 'softfail');
-        return $ret;
+    my ($script_name, %args) = @_;
+    my $script_url = $args{script_url} // data_url("virt_autotest/$script_name");
+    my $machine = $args{machine} // 'localhost';
+
+    my $cmd = "curl -s -o ~/$script_name $script_url";
+    $cmd = "ssh root\@$machine " . "\"$cmd\"" if ($machine ne 'localhost');
+    unless (script_retry($cmd, retry => 2, die => 0) eq 0) {
```

Comment:
I think "== 0" is better, because "eq 0" is for string comparison.

Reply:
ok